cpp/tests/fixed_point/fixed_point_tests.cpp

```diff
@@ -53,6 +53,7 @@ TYPED_TEST(FixedPointTestAllReps, SimpleDecimalXXConstruction)
   auto num4 = cudf::convert_floating_to_fixed<decimalXX>(1.234567, scale_type(-4));
   auto num5 = cudf::convert_floating_to_fixed<decimalXX>(1.234567, scale_type(-5));
   auto num6 = cudf::convert_floating_to_fixed<decimalXX>(1.234567, scale_type(-6));
+  auto num7 = cudf::convert_floating_to_fixed<decimalXX>(0.0, scale_type(-4));
```

Comment:
It appears all of these tests only test the host code and not the device code?
There is host/device code logic divergence here:
https://github.com/rapidsai/cudf/pull/15905/files#diff-b5b5c5e5f6104cada46b6dbe39765dd0d70319c798653242a53024b774399524R50

Reply:
GPU-specific tests are [performed here](https://github.com/rapidsai/cudf/blob/d9a3728d37e0223afd9cfa525bd7ac8b43b39e63/cpp/tests/unary/cast_tests.cpp#L653) with the cudf::cast() kernel.